clang/tools/clang-shlib/CMakeLists.txt

```diff
@@ -48,11 +48,13 @@ add_clang_library(clang-cpp
                   ${_OBJECTS}
                   LINK_LIBS
                   ${_DEPS})
+# AIX linker does not support version script
+if (NOT ${CMAKE_SYSTEM_NAME} MATCHES "AIX")
+  configure_file(simple_version_script.map.in simple_version_script.map)
 
-configure_file(simple_version_script.map.in simple_version_script.map)
-
-if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
```

Comment:
This implies `NOT ${CMAKE_SYSTEM_NAME} MATCHES "AIX"`, hence the `target_link_options` is already unreachable.

Reply:
You are right, @aaronpuchert! With commit https://github.com/llvm/llvm-project/commit/eaa0a21d21962280dc2c03a09152510f6162a576 the AIX build problem is fixed already so the change in https://github.com/llvm/llvm-project/pull/117342 does not make sense any more. I created PR [Revert "[AIX] Fix AIX BuildBot failure as AIX linker doesn't support version script."](https://github.com/llvm/llvm-project/pull/117444).